wordSearchSolver.py

```diff
@@ -49,65 +49,64 @@ def get_input():  # get input two letters
 
 
 def analyze(user_input):
-    answerCoordinates = []
 
-    def addCoordinates(originalCoordinate, displacement):
-        return tuple(map(sum, zip(originalCoordinate, displacement)))
+    def add_coordinates(original_coord, displacement):
+        return tuple(map(sum, zip(original_coord, displacement)))
 
-    def validCoordinate(coordinateToCheck):
-        if all(((coordinateToCheck[0] >= 0), (coordinateToCheck[0] < (GRID_DIMENSION[0] + 1)),
-                (coordinateToCheck[1] >= 0), (coordinateToCheck[1] < (GRID_DIMENSION[0] + 1)))):
-            return (True, coordinateToCheck)
+    def valid_coord(coord_to_check):
+        if all(((coord_to_check[0] >= 0), (coord_to_check[0] < (GRID_DIMENSION[0] + 1)),
+                (coord_to_check[1] >= 0), (coord_to_check[1] < (GRID_DIMENSION[0] + 1)))):
+            return True, coord_to_check
         else:
-            return (False, None)
+            return False, None
 
     def first_letter():  # find all coordinates that contain first letter
         first_letter_coord = []
         first_letter = user_input[0]
         for item in word_grid.items():  # if matching letter, add coordinates to list
             if item[1] == first_letter:
                 first_letter_coord.append(item[0])
-        logging.debug("firstLetter(): {}".format(first_letter_coord))
+        logging.debug("first_letter(): {}".format(first_letter_coord))
         return first_letter_coord
 
     def second_letter(first_letter_coord):
-        combinationsDict = {}
-        for coordinateToCheck in first_letter_coord:
+        combinations_dict = {}
+        for coordinate_to_check in first_letter_coord:
             # search all possible second letter combinations
-            coordinatesToCheck = [(-1, 1), (0, 1), (1, 1), (-1, 0), (1, 0), (1, -1), (0, -1), (-1, -1)]
-            for modCoordinate in coordinatesToCheck:
-                validCoordinateOutput = validCoordinate(addCoordinates(coordinateToCheck, modCoordinate))
+            coordinates_to_check = [(-1, 1), (0, 1), (1, 1), (-1, 0), (1, 0), (1, -1), (0, -1), (-1, -1)]
+            for mod_coordinate in coordinates_to_check:
+                valid_coordinate_output = valid_coord(add_coordinates(coordinate_to_check, mod_coordinate))
                 # if a valid coordinate
-                if validCoordinateOutput[0]:
+                if valid_coordinate_output[0]:
                     # check if there is a match. If there is one next to first letter, append and break loop
-                    if word_grid[validCoordinateOutput[1]] == user_input[1]:
-                        combinationsDict.setdefault(coordinateToCheck, []).append(
-                            {"coordinate": validCoordinateOutput[1], "modCoordinate": modCoordinate})
-                logging.debug(
-                    "NumPossibilities: {}\nsecond_letter(): {}".format(len(combinationsDict.keys()), combinationsDict)) # combinationsDict.keys() does not work here
```

Comment:
finding the len(combination_dict.keys()) does not actually work. Doesn't affect overall file performance, so not sure what to do.